### DISCUSSION TITLE
DevTools in release mode

### DISCUSSION THREAD

**Discussion by rdwnsjjd** [tauri-apps/tauri#3059] [Q&A] [answered]
How to enable dev tools (inspect) in the final built (released) app?
I used yarn tauri build --debug but still not showing inspect.

**Comment by FabianLars** [comment, marked as answer]
This got answered on discord. For others who are searching for this here is the answer:
There are two methods:
1) Build the whole app in debug mode (`tauri build --debug`). **The binary will end up in `target/debug`!** This of course is not actually in release mode, but the bundler etc. will do the same as they would in the actual release mode. The only difference is how rust compiles the binaries.
2) Build the app in release mode but `wry` in debug mode. **The binary will end up in `target/release`!** For this you need to add the following to your `Cargo.toml` file. Edit: This answer was posted for 1.0.0-beta versions, for rc and beyond see the "Edit" section at the end of this comment.
```toml
[profile.release.package.wry]
debug = true
debug-assertions = true
```

----

Edit: As of recent versions it is possible to add the `"devtools"` features flag to the tauri dependency in Cargo.toml to enable the dev tools in release mode without debug mode shenanigans. Note, that we of course still don't recommend that you ship your app with devtools enabled!
Example: `tauri = { version = "1.0.0-rc.8", features = ["api-all", "devtools"] }`

**Reply by Lantianyou** [reply]
This is not working for tauri 1.0, would you kindly share any solutions used in tauri@1.0-rc

**Reply by FabianLars** [reply]
@Lantianyou You can check the "edit" section at the bottom of my message about the `devtools` feature flag and this comment for an example: <https://github.com/tauri-apps/tauri/discussions/3059#discussioncomment-2691931>.
Edit: I edited my message to include the same example.

**Reply by Lantianyou** [reply]
Thank you so much about it. It is my fault I have not read your detailed explaination.

**Comment by TheButlah** [comment]
Just tried this, using the `wry` approach worked with cargo run but not `npm run tauri build`. I had to add the `devtools` feature flag in the `Cargo.toml` to get it to work with all ways of building.

**Comment by liudonghua123** [comment]
I added the following to `src-tauri/Cargo.toml`, and run `yarn tauri build -t x86_64-pc-windows-msvc`, but I still could not open `devtools`.

```toml
[profile.release.package.wry]
debug = true
debug-assertions = true
```

**Reply by liudonghua123** [reply]
I can use `devtools` via updating the dependences from `tauri = { version = "1.0.0-rc.7", features = ["api-all"] }` to `tauri = { version = "1.0.0-rc.7", features = ["api-all", "devtools"] }` in `src-tauri/Cargo.toml`.

**Comment by gwennlbh** [comment]
hi huuuuuuuh seems like the feature has been removed in v2? Is it possible to add the feature somehow ?

**Reply by FabianLars** [reply]
It still works the same in v2. Just add the `"devtools"` flag to the tauri dependency in Cargo.toml.

If you have issues, please tell us what kind of issues :)

**Reply by gwennlbh** [reply]
hiiii hum yeh it works i just added the flag to tauri-build too by mistake

thx for the quick rely :)

**Reply by gknapp** [reply]
@FabianLars is it possible to control the console / chrome inspector display at runtime? For example, if a certain CLI parameter is passed it will  open at app start, otherwise it doesn't show. Or is it a build time only configuration?

**Reply by FabianLars** [reply]
there's https://docs.rs/tauri/latest/tauri/webview/struct.WebviewWindowBuilder.html#method.devtools

**Comment by 80avin** [comment]
For those trying with `npm run tauri build`, the command has to be 
```sh
npm run tauri build -- --debug
```

**Comment by peabnuts123** [comment]
The correct, canonical way to do this is to add the `devtools` feature flag to your tauri dependency (works v1 and v2). Don't publish your entire app in debug mode.

```toml
[dependencies]
tauri = { version = "2", features = ["devtools"] }

```
